server/service/appconfig.go

```diff
@@ -98,15 +98,13 @@ func getAppConfigEndpoint(ctx context.Context, request interface{}, svc fleet.Se
 		return nil, err
 	}
 
-	var smtpSettings fleet.SMTPSettings
-	var ssoSettings fleet.SSOSettings
-	var hostExpirySettings fleet.HostExpirySettings
+	// Only the Global Admin should be able to see see SMTP, SSO and osquery agent settings.
```

Comment:
Correct me if I'm wrong, but this is a breaking change for the API, right? I.e. non-admin users could see those settings before? I guess it's ok if it was considered a bug/security issue.

Oh actually I think it was returning those objects as empty, while now they won't be returned at all. Which is still technically a breaking change, but quite minor (can certainly be considered a bug fix, which it is as per the related issue).

Reply:
Good point.

The whole /config endpoint needs a revamp for sure. (Fleet v5 👀.)